Find a functionally equivalent code:android.provider.DocumentsProvider.openFile:COMMENT
Method Modifier: public      final       
Comment:/**
 * Implementation is provided by the parent class. Cannot be overriden.
 *
 * @see #openDocument(String, String, CancellationSignal)
 */

@@ -1,3 +1,4 @@
 {
+    enforceTree(uri);
     return openDocument(getDocumentId(uri), mode, signal);
 }

